src/ejabberd_tls.erl

```diff
@@ -0,0 +1,58 @@
+-module(ejabberd_tls).
+-author('denys.gonchar@erlang-solutions.com').
+
+%% tls interfaces required by ejabberd_socket & ejabberd_receiver modules.
+-export([ tcp_to_tls/2,
+          send/2,
+          recv_data/2,
+          controlling_process/2,
+          sockname/1,
+          peername/1,
+          setopts/2,
+          get_peer_certificate/1,
+          close/1 ]).
+
+-export([get_sockmod/1]).
+
+-record(ejabberd_tls_socket, { tls_module,
+                               tcp_socket,
+                               tls_socket
+                             }).
+
+-define(FN_WRP1(FN),FN(#ejabberd_tls_socket{tls_module = M, tls_socket = S}) -> M:FN(S)).
+-define(FN_WRP2(FN),FN(#ejabberd_tls_socket{tls_module = M, tls_socket = S},P1) -> M:FN(S,P1)).
+
+tcp_to_tls(TCPSocket,Opts) ->
+  Module = proplists:get_value(tls_module, Opts, fast_tls),
```

Comment:
do we support anything else?

Reply:
sure fast_tls and just_tls (based on the standard ssl implementation)

Reply:
So, here should be a comment:

```
Module can be any erlang module, that implements `NAME_IT` behaviour.
```